File: cs445/threading/java/src/gus/basic/ActsLikeThread.java
package gus.basic;

// java -cp bin gus.basic.ActsLikeThread
public class ActsLikeThread extends Thread {
  private String message = null;
  private long sleepTimeMs = 0;

  public ActsLikeThread(String message, long sleepTimeMs) {
    this.message = message;
    this.sleepTimeMs = sleepTimeMs;
  }

  public void run() {
    System.out.println("started " + message);
    try {
      while(true) {
        System.out.println(message);
        sleep(sleepTimeMs);
      }
    } catch (InterruptedException e) {}
    System.out.println("stopped " + message);
  }

  public static void main(String [] args) {
    Thread t1 = new ActsLikeThread("Hello", 500);
    Thread t2 = new ActsLikeThread("world", 250);
    t1.start();
    t2.start();

    System.out.println("done with main");
    try {Thread.sleep(5000);} catch (InterruptedException e) {}
    t1.interrupt();
    t2.interrupt();
  }
}

File: cs445/threading/ruby/basic/acts_like_thread.rb
#
# ruby basic/acts_like_thread.rb
class ActsLikeThread < Thread
  def initialize(message, sleep_time)
    @message = message
    @sleep_time = sleep_time

    super do
      loop do
        puts @message
        sleep(@sleep_time)
      end
    end
  end
end # ActsLikeThread

t1 = ActsLikeThread.new("Hello", 0.25);
t2 = ActsLikeThread.new("world", 0.5);

puts "Done starting threading"
sleep(5)
puts "Awake now"

t1.kill; t2.kill
